src/app_main.rs

```diff
@@ -75,72 +79,31 @@ impl<T: Send + 'static, V: View<T> + 'static> AppLauncher<T, V> {
         let mut main_state = MainState::new(self.app, window);
 
         event_loop
-            .run(move |event, elwt| match event {
-                winit::event::Event::WindowEvent {
-                    event: WindowEvent::CloseRequested,
-                    ..
-                } => elwt.exit(),
-                winit::event::Event::WindowEvent {
-                    event: WindowEvent::RedrawRequested,
-                    ..
-                } => main_state.paint(),
-                winit::event::Event::WindowEvent {
-                    event: WindowEvent::Resized(winit::dpi::PhysicalSize { width, height }),
-                    ..
-                } => main_state.size(Size {
-                    width: width.into(),
-                    height: height.into(),
-                }),
-                winit::event::Event::WindowEvent {
-                    event: WindowEvent::ModifiersChanged(modifiers),
-                    ..
-                } => {
-                    let mut m = Modifiers::empty();
-                    let ms = modifiers.state();
-                    if ms.contains(winit::keyboard::ModifiersState::SHIFT) {
-                        m |= Modifiers::SHIFT;
-                    }
-                    if ms.contains(winit::keyboard::ModifiersState::CONTROL) {
-                        m |= Modifiers::CONTROL;
-                    }
-                    if ms.contains(winit::keyboard::ModifiersState::SUPER) {
-                        m |= Modifiers::SUPER;
-                    }
-                    if ms.contains(winit::keyboard::ModifiersState::ALT) {
-                        m |= Modifiers::ALT;
-                    }
-                    main_state.mods(m);
-                }
-                winit::event::Event::WindowEvent {
-                    event:
+            .run(move |event, elwt| {
+                if let winit::event::Event::WindowEvent { event: e, .. } = event {
```

Comment:
Just a thing/tip I like to do to, to avoid intense nesting for cases like this:

```suggestion
                let winit::event::Event::WindowEvent { event: e, .. } = event else {
                    return;
                };
```

Reply:
The reason it's like this is because clippy was complaining about match with a single pattern. There will be cases where we need to handle things other than `WindowEvent` and this will return to being a `match`